File: Examples/Test_Error/Test_Error.m
clear;
clc;

disp("Error 1");
err1 = Error('code', 20);
err1.Summary();
disp("Error 2");
err2 = Error("filename", "Test_Error.cpp", 'methodname', "main");
err2.Summary();
err21 = Error('methodname',"main");
err21.Summary();
err22 = Error("filename", "Test_Error.cpp");
err22.Summary();
err23 = Error();
err23.Summary();
disp("Error 3");
err3 = Error("filename", "Test_Error.cpp", 'classname', "Main", 'methodname', "main");
err3.Summary();
disp("Error 4");
err4 = Error('pathprefix', "/src", "filename", "Test_Error.cpp", 'classname', "Main", 'methodname', "main");
err4.Summary();

disp("Error 6");
err6 = Error('pathprefix', "/src/", "filename", "Test_Error.cpp", 'classname', "Main", 'methodname', "main(");
err6.SetType2MemoryShortage();
err6.Summary();

disp("Error 7");
err7 = Error('pathprefix', "/src/", "filename", "Test_Error.cpp", 'classname', "Main",  'methodname', "main(", 'msg', "message");
err7.SetType2MemoryShortage();
err7.Summary();

File: Examples/Test_Error/Test_Error.cpp
// Test_Error.cpp : Defines the entry point for the application.
//
#include <iostream>
#include "../../Error.h"

using namespace std;
using namespace Toolbox;

int main()
{
	cout << "Error 1" << endl;
	Error err1(TOOLBOX_ERR_INTERNAL);
	err1.Summary();
	cout << "\nError 2" << endl;
	Error err2(NULL, "Test_Error.cpp", "main");
	err2.Summary();
	Error err21(NULL, NULL, "main");
	err21.Summary();
	Error err22(NULL, "Test_Error.cpp", NULL);
	err22.Summary();
	Error err23(NULL, NULL, NULL);
	err23.Summary();
	cout << "\nError 3" << endl;
	Error err3(NULL, "Test_Error.cpp", "Main", "main");
	err3.Summary();
	cout << "\nError 4" << endl;
	Error err4("/src", "Test_Error.cpp", "Main", "main");
	err4.Summary();

	cout << "\nError 6" << endl;
	Error err6("/src/", "Test_Error.cpp", "Main", "main(");
	err6.SetType2MemoryShortage();
	err6.Summary();

	cout << "\nError 7" << endl;
	Error err7("/src/", "Test_Error.cpp", "Main", "main(", "message");
	err7.SetType2MemoryShortage();
	err7.Summary();
	return 0;
}
